Example/PaymentSheet Example/PaymentSheet Example/PaymentSheetTestPlayground.swift

```diff
@@ -704,9 +704,21 @@ extension PaymentSheetTestPlayground {
     func confirmHandlerForServerSideConfirmation(_ paymentMethodID: String,
                                                  _ shouldSavePaymentMethod: Bool,
                                                  _ intentCreationCallback: @escaping (Result<String, Error>) -> Void) {
-        enum ServerSideConfirmationError: Error {
+        enum ServerSideConfirmationError: Error, LocalizedError {
```

Comment:
This needs to be a `LocalizedError` and needs to implement `errorDescription: String?` for us to be able to read the [localizedDescription](https://developer.apple.com/documentation/foundation/nserror/1414418-localizeddescription) in PaymentSheet. 

Should we update the intent creation callback to be a `LocalizedError` instead of just an `Error`?

Reply:
We read localizedDescription here: https://github.com/stripe/stripe-ios/blob/7791e585d7f08cbdcec029496536c8d97c99d2ab/StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Error%2BPaymentSheet.swift#L21

Reply:
Hmm, looks like we can't use `LocalizedError` with `Result`
https://developer.apple.com/documentation/swift/result

![CleanShot 2023-03-20 at 11 34 48](https://user-images.githubusercontent.com/88012362/226421653-98370e14-5d2c-42e9-87a6-81016a1520aa.png)

Reply:
I think we either introduce our own new type for errors so they are displayed nicely or find a way to document this.

Reply:
I don't think we need to **require** merchants use a LocalizedError. Also, NSError doesn't conform to it (confusingly). https://figure.ink/2021/07/18/practical-localized-error.html

Adding documentation sounds good.